helpers/setupjsongenerator.py

```diff
@@ -0,0 +1,675 @@
+        """
+        self._valid_storagerouter(storagerouter_ip)
+        Toolbox.verify_required_params(required_params={'name': (str, None, True)}, actual_params={'name': name}, verify_keys=True)
+
+        if name not in self._domains:
+            raise ValueError('Invalid domain passed: {0}'.format(name))
+
+        path = self.config['storagerouters'][storagerouter_ip]
+        if 'domains' not in path.keys():
+            path['domains'] = {}
+        path = path['domains']
+        config_key = 'domain_guids' if recovery is False else 'recovery_domain_guids'
+        if config_key not in path:
+            path[config_key] = []
+        path[config_key].append(name)
+
+    def remove_domain_from_sr(self, storagerouter_ip, name):
+        """
+        Remove the given domain from the storagerouter
+        :param storagerouter_ip: storagerouter to remove the domains from
+        :type storagerouter_ip: str
+        :param name: name of the domain to remove
+        :type name: str
+        """
+        try:
+            _ = self.config['storagerouters'][storagerouter_ip]['domains']['domain_guids']
+            _.remove(name)
+        except Exception:
+            pass
+
+    def add_backend(self, backend_name, domains=None, scaling='LOCAL'):
+        """
+        Add a backend with provided domains and scaling to the model.
+        :param backend_name: name of the backend
+        :type backend_name: str
+        :param domains: domains the backend is linked to
+        :type domains: {}
+        :param scaling:
+        :type scaling: str
+        """
+        if domains is None:
+            domains = []
+        else:
+            for domain_name in domains:
+                if domain_name not in self._domains:
+                    raise ValueError('Invalid domain passed: {0}'.format(domain_name))
+
+        Toolbox.verify_required_params(required_params={'backend_name': (str, Toolbox.regex_backend, True),
+                                                        'domains': (list, self._domains, True),
+                                                        'scaling': (str, AlbaBackend.SCALINGS, True)},
+                                       actual_params={'backend_name': backend_name,
+                                                      'domains': domains,
+                                                      'scaling': scaling}, verify_keys=True)
+
+        be_dict = {'name': backend_name,
+                   'domains': {'domain_guids': domains},
+                   'scaling': scaling}
+
+        self._backends.append(be_dict['name'])
+        if 'backends' not in self.config:
+            self.config['backends'] = []
+        self.config['backends'].append(be_dict)
+
+    def remove_backend(self, backend_name):
+        """
+        Remove backend with given name from model
+        :param backend_name: name of the backend to remove
+        :type backend_name: str
+        """
+        for backend in self.config['backends']:
```

Comment:
Since you are looping here anyway, use enumerate(self.config['backends']) to get the index to avoid the .index(backend) (which loops again)